external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -1125,8 +1129,16 @@ fn parse_attributes(context: &mut Context) -> Result<Vec<Attributes>, Box<Diagno
             start_loc,
             end_loc,
             attributes_,
-        ))
+        ));
+        if let Some(new_doc_comments) = context.tokens.read_doc_comments() {
+            doc_comments.push(new_doc_comments);
+        }
     }
+
+    if !doc_comments.is_empty() {
+        context.tokens.attach_doc_comments(doc_comments.join("\n"));
+    }
```

Comment:
It may be worth noting that this code generally attaches the doc comment to the location of the first modifier, if any. This isn't a problem per say, but may make it harder to find that location in a few cases.